Add link to requirement levels on Markdown table headers

lmolkova opened the pull request:
Apply suggestions from https://github.com/open-telemetry/semantic-conventions/pull/283

Oberon00 commented on semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py:
```diff
@@ -169,14 +177,14 @@ def to_markdown_attr(
     def derive_requirement_level(self, attribute: SemanticAttribute):
         if attribute.requirement_level == RequirementLevel.REQUIRED:
-            required = "Required"
+            required = "`Required`"
```
Why use code formatting here? It's not code(-like) IMHO.

lmolkova replied:
The motivation is provided here: https://github.com/open-telemetry/semantic-conventions/pull/283#discussion_r1361413434

people not deeply familiar with otel assume that the requirement level is plain English. Having formatting emphasizes it.

Oberon00 commented on semantic-conventions/src/opentelemetry/semconv/main.py:
```diff
@@ -214,6 +215,14 @@ def add_md_parser(subparsers):
         dest="md_enable_deprecated",
         action="store_false",
     )
+    parser.add_argument(
```
This is a (preparation for a) major change IMHO, more than a "random editorial improvement". I like it though, it could be the first step towards better links, including from code generators.

lmolkova replied:
what's the concern here? The PR title?

joaopgrassi replied:
I changed the title, to align with the changelog entry. I think that resolves this.

Oberon00 commented on semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py:
```diff
@@ -80,7 +80,6 @@ class MarkdownRenderer:
     prelude = "<!-- semconv {} -->\n"
-    table_headers = "| Attribute  | Type | Description  | Examples  | Requirement Level |\n|---|---|---|---|---|\n"
     table_headers_omitting_req_level = (
         "| Attribute  | Type | Description  | Examples  |\n|---|---|---|---|\n"
```
"Type" could also be linked to the list of attribute types.

lmolkova replied:
Added the link. We don't have a dedicated section that described attribute type, so added link to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute. Totally open to revert this change.

joaopgrassi replied:
I think that is good enough for me!

joaopgrassi commented:
@lmolkova I took the liberty to update this PR and addressed the issue with the PR title. What is else blocking in merging this?

lmolkova commented:
> @lmolkova I took the liberty to update this PR and addressed the issue with the PR title. What is else blocking in merging this?

thanks a lot!!

lmolkova commented:
@open-telemetry/specs-semconv-maintainers can it be merged?

Oberon00 commented on semantic-conventions/src/opentelemetry/semconv/main.py:
```diff
@@ -214,6 +215,14 @@ def add_md_parser(subparsers):
     )
+    parser.add_argument(
+        "--specification-version",
+        help="Set specification version to use.",
```
To manage expectations, maybe mention that this does not yet support managing arbitrary internal links (although I guess this might then be forgotten even if we implement that)

lmolkova replied:
I clarified the comment.

lmolkova commented:
@open-telemetry/specs-semconv-maintainers can it be merged?

joaopgrassi commented:
@lmolkova I think we still need to address the comments about the link to the spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-requirement-level.md. That is moved now to semconv. 

But if we change the link to the semconv, then other problems appear as I have noted above. 

Another idea is to just link to the website instead

lmolkova commented:
@joaopgrassi great catch and great suggestion!

I changed req level link to use website. We don't have a clear link to attribute types, so I removed link for types.
I think it's ready (yet again)

joaopgrassi commented on semantic-conventions/CHANGELOG.md:
```diff
@@ -6,6 +6,8 @@ Please update the changelog as part of any significant pull request.
 
 - BREAKING: Make stability and deprecation independent properties.
   ([#244](https://github.com/open-telemetry/build-tools/pull/244))
+- Add link to attribute types and requirement levels from Markdown table title.
```
```suggestion
- Add link requirement levels from Markdown table title.
```

Also the PR title now is outdated.

joaopgrassi commented on semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py:
```diff
@@ -100,6 +96,27 @@ def __init__(
+            f"{options.specification_repo_tag}"
+        )
+        attribute_req_level_url = (
+            f"{spec_root_url}/specification/common/attribute-requirement-level.md"
```
I had this but didn't publish 🤦. Just ignore.
